TYPO3.Flow/Classes/TYPO3/Flow/Cache/Backend/RequireOnceFromValueTrait.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace TYPO3\Flow\Cache\Backend;
+
+/*
+ * This file is part of the TYPO3.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use TYPO3\Flow\Cache\Exception as CacheException;
+
+/**
+ * RequireOnceFromValueTrait
+ */
+trait RequireOnceFromValueTrait
+{
+    /**
+     * @var array
+     */
+    protected $_requiredEntryIdentifiers = [];
+
+    /**
+     * Loads PHP code from the cache and require_onces it right away.
+     *
+     * @param string $entryIdentifier An identifier which describes the cache entry to load
+     * @return mixed Potential return value from the include operation
+     * @api
+     */
+    public function requireOnce($entryIdentifier)
+    {
+        $value = trim($this->get($entryIdentifier));
+        if ($value === '') {
+            return false;
+        }
+        $hash = md5($value);
+        if (isset($this->_requiredEntryIdentifiers[$hash])) {
```

Comment:
If you ensure the "once" with this, I guess it should be `include` afterwards. Otherwise there is some overhead from PHP (trying to) check the file for being included already… no?